Topic/gitattributes

paulp opened the pull request.

paulp committed:
New .gitattributes file.

This should assist in keeping line endings straight.
It is designed to enforce LF endings everywhere except
for files specifically for windows.

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/463/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1170/

paulp commented:
Review by @jsuereth

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/463/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1170/

paulp commented:
Yeah, oversight.  I added that.  Renormalizing didn't change any of them though.

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/476/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1184/

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/476/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1184/

gkossakowski commented:
This commit is huge but it only modifies `*.check` files so it's unlikely to cause nasty merge conflicts for people busy with getting 2.10.x out. @jsuereth: can you lgtm?

jsuereth commented:
Let me validate on windows first, sorry.  That machine is dog slow.

retronym commented on .gitattributes:
```diff
@@ -0,0 +1,25 @@
+*.check   text
+
+# Windows-specific files get windows endings
+*.bat     eol=crlf
```
*.cmd

paulp commented:
PLS REBUILD ALL

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/509/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1218/

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/509/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1218/

paulp commented:
@jsuereth ping / bump

paulp committed:
Normalized line endings.

This brings all the files into line with the .gitattributes
settings, which should henceforth be automatically maintained
by git.

paulp commented:
I rebased it on 2.10.x.  This thing obviously drifts pretty quickly so what say we merge it.

xeno-by commented:
A smoke test shows that this works okishly on a windows machine. Ant works, bat files from pack work, partest.bat works as well. I'd say we merge

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/549/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1257/

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/549/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1257/

jsuereth commented:
Sorry, forgot to check on ye ole windows,   looks OK from the peanut
gallery too.
